src/compiler/program.ts

```diff
@@ -991,7 +991,13 @@ namespace ts {
         return sortAndDeduplicateDiagnostics(diagnostics);
     }
 
-    export function formatDiagnostics(diagnostics: Diagnostic[], host: CompilerHost): string {
+    export interface FormatDiagnosticsHost {

```

Comment:
Can you please clarify this interface? Thanks a lot!

It's a contract between what and what?
Who is meant to provide the implementation?
Do existing host scenarios need to do anything?
Maybe a use case please?

Sorry for being inquisitive, but there are a few other `*Host` interfaces at various levels of abstraction, it becomes harder to follow. Please give us a picture why `FormatDiagnosticsHost` is there, beyond self-evident 'to deal with diagnostic formatting'. I am sure others will benefit from the same, especially where TS is an embedded library.

Many thanks!